UnityProject/Assets/Scripts/Health/Sickness/Contagion.cs

```diff
@@ -0,0 +1,58 @@
+﻿using UnityEngine;
+using UnityEngine.EventSystems;
+
+namespace Assets.Scripts.Health.Sickness
+{
+	public class Contagion: MonoBehaviour
+	{
+		public Sickness Sickness;
+
+		[SerializeField]
+		[Tooltip("Time (in seconds) for the contagion to despawn itself")]
+		private int contagionTime = 20;
+
+		private float spawnedTime;
+
+		private RegisterTile registerTile;
+		
+		public void Start()
+		{
+			spawnedTime = Time.time;
+		}
+
+		public void Update()
```

Comment:
Might want to change this to use the update manager

Reply:
Please tell me more about the Update Manager.  What does it do?

Reply:
Example:
https://github.com/unitystation/unitystation/blob/a3b4b0cd4af6328d280d74635325b71695bc239f/UnityProject/Assets/Scripts/Items/Others/ItemPinpointer.cs#L208

Reply:
Now uses the UpdateManager.  It's easy enough to use! :) Thanks.